src/plugins/unified_search/public/dataview_picker/change_dataview.tsx

```diff
@@ -111,8 +111,9 @@ export function ChangeDataView({
       const savedDataViewRefs: DataViewListItemEnhanced[] = savedDataViews
         ? savedDataViews
         : await data.dataViews.getIdsWithTitle();
+      // not propagate the adHoc dataviews on the list for text based languages
       const adHocDataViewRefs: DataViewListItemEnhanced[] =
-        adHocDataViews?.map(mapAdHocDataView) || [];
+        (!isTextBasedLangSelected && adHocDataViews?.map(mapAdHocDataView)) || [];
```

Comment:
The problem with this check is that it completely removes all `adHocDataViews` without any distinction between the text based ones and those created via flyout by the user:
![delete_all_adhoc_views](https://github.com/elastic/kibana/assets/924948/d442668b-65dd-44de-8bcd-439eaa06fac4)

Reply:
Yes this is fine. We are going to remove the ESQL from the dataview picker. It will be a switch instead and we will always default to the index pattern of the default dataview. So we want all adhoc dataviews to be lost. 

In general, the logic is no dataviews on text based language mode.